src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,39 @@
+import { createContext, useContext, useState } from 'react'
+
+interface AuthContextType {
+  user: string | null
+  login: (email: string) => void
+  logout: () => void
+}
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined)
+
+export function AuthProvider({ children }: { children: React.ReactNode }) {
+  const [user, setUser] = useState<string | null>(() => {
+    return localStorage.getItem('user')
+  })
+
+  const login = (email: string) => {
+    localStorage.setItem('user', email)
+    setUser(email)
+  }
+
+  const logout = () => {
+    localStorage.removeItem('user')
+    setUser(null)
+  }
```

Comment:
`useStorageState` 라는 훅을 하나 만들어서 사용해주면 어떨까요?
```suggestion
  const [user, setUser] = useStorageState<string | null>('user');
  const login = (email: string) => setUser(email);
  const logout = () => setUser(null);
```

Reply:
useStorageState 훅 추가 및 적용했습니다.